tests/test_render.py

```diff
@@ -1898,6 +1898,79 @@ def update_output(*args):
         self.assertEqual(call_count.value, 4)
         self.percy_snapshot('button-2 click again')
 
+    def test_request_hooks(self):
+        app = Dash(__name__)
+
+        app.index_string = '''
+        <!DOCTYPE html>
+        <html>
+            <head>
+                {%metas%}
+                <title>{%title%}</title>
+                {%favicon%}
+                {%css%}
+            </head>
+            <body>
+                <div>Testing custom DashRenderer</div>
+                {%app_entry%}
+                <footer>
+                    {%config%}
+                    {%scripts%}
+                    <script id="_dash-renderer" type"application/json">
+                        const renderer = new DashRenderer({
+                            request_pre: () => {
```

Comment:
Can we use the arguments to these hooks in this test? Just to ensure that the right things are there and that they don't change unintentionally.

Reply:
Yeah good call. I've added them and I'm doing an assertion on the output. However, if this output ever changes, this test will automatically fail. Are we ok with that?

Reply:
> if this output ever changes, this test will automatically fail. Are we ok with that?

In fact that's exactly why we want this in the tests - so we know it changed and we can let custom hooks users know they will need to update their code!